[JIT] Refactor SchemaInfo training ops to account for rrelu_with_noise case

ttanpcs opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82442
* __->__ #82441

- Refactor SchemaInfo to be able to handle cases where other variables besides running_mean and running_var mutate due to training = true
- Add special case rrelu_with_noise to fix https://github.com/pytorch/pytorch/issues/82434
- Tested by running SchemaInfo tests

Commit:
[JIT] Refactor SchemaInfo training ops to account for rrelu_with_noise case

[ghstack-poisoned]

davidberard98 commented on torch/csrc/utils/schema_info.cpp:
```diff
@@ -262,24 +263,38 @@ std::vector<c10::FunctionSchema> SchemaInfo::getNonDeterministicOps() {
+        {"running_mean", "running_var"}},
+       {"aten::rrelu_with_noise(Tensor self, Tensor noise, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=False, Generator? generator=None) -> Tensor",
+        {"noise"}},
+       {"aten::rrelu_with_noise.out(Tensor self, Tensor(b!) noise, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=False, Generator? generator=None, *, Tensor(a!) out) -> Tensor(a!)",
+        {"noise"}},
+       {"rrelu_with_noise_(Tensor(a!) self, Tensor(b!) noise, Scalar lower=0.125, Scalar upper=0.3333333333333333, bool training=False, Generator? generator=None) -> Tensor(a!)",
```
should these not have (b!) annotating the noise param? I think this change hasn't been added to native_functions.yaml yet

goldenxuett replied:
Oh I must've copy pasted the wrong version of the schemas. I'll change this.

davidberard98 reviewed (commented):
looks good, but had a question about the schemas annotated with `(b!)`

also - if possible a test would be great, especially to check the inplace/out variant cases explicitly (not sure if it would be covered in the opinfo test)

Commit:
Update on "[JIT] Refactor SchemaInfo training ops to account for rrelu_with_noise case"


- Refactor SchemaInfo to be able to handle cases where other variables besides running_mean and running_var mutate due to training = true
- Add special case rrelu_with_noise to fix https://github.com/pytorch/pytorch/issues/82434
- Tested by running SchemaInfo tests

[ghstack-poisoned]

Commit:
Update on "[JIT] Refactor SchemaInfo training ops to account for rrelu_with_noise case"


- Refactor SchemaInfo to be able to handle cases where other variables besides running_mean and running_var mutate due to training = true
- Add special case rrelu_with_noise to fix https://github.com/pytorch/pytorch/issues/82434
- Tested by running SchemaInfo tests

[ghstack-poisoned]

davidberard98 reviewed (approved):
looks good :+1:

goldenxuett commented:
@pytorchbot merge
